src/caffe/common.cpp

```diff
@@ -192,6 +201,26 @@ void Caffe::DeviceQuery() {
   return;
 }
 
+bool Caffe::CheckDevice(const int device_id) {
+  bool r =  ((cudaSuccess == cudaSetDevice(device_id)) &&
+            (cudaSuccess == cudaFree(0)));
+  cudaGetLastError();
```

Comment:
Presumably this is to reset any error that may have occurred; if so that's worthy of a comment, since right now it looks like we're just discarding the error.